### PR TITLE
DEPR: indexing Series with single-entry list

### LINKED ISSUE #31299: DEPR: disallow series[[slice(...)]]
One of the more convoluted parts of `Series.__getitem__` is for handling single-entry lists that contain a slice, where we end up just returning `self[key[0]]`.  We have two tests that hit this path, but it isn't clear to me why this should be supported.

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #31299 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Could use a better warning message, suggestions welcome.

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -928,6 +928,17 @@ def _get_with(self, key):
                 # [slice(0, 5, None)] will break if you convert to ndarray,
                 # e.g. as requested by np.median
                 # FIXME: hack
+                if isinstance(key, list):
+                    # GH#31299
+                    warnings.warn(
+                        "Indexing with a single-item list containing a "
+                        "slice is deprecated and will raise in a future "
```
Comment (TomAugspurger) [inline pandas/core/series.py]: Do we know what it will raise? Or will it depend on the value?

**Review by TomAugspurger — COMMENTED** [review]
This seems reasonable. Will users see this warning in addition to the warnings in https://github.com/pandas-dev/pandas/issues/31299#issuecomment-578418088?

Will need a whatsnew note.

File: pandas/core/series.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/series.py]: Best guess is TypeError, depends on where we let it fall through

**Comment by jbrockmendel** [comment]
> Will users see this warning in addition to the warnings in #31299 (comment)?

No, this will change the list to tuple before calling `ndarray.__getitem__`

> Will need a whatsnew note.

updated

**Comment by jreback** [comment]
thanks, can you add to the deprecation issue (do this for anything we are deprecating).

**Pull request merged**